internal/controller/controllers/agent_controller.go

```diff
@@ -348,6 +363,35 @@ func (r *AgentReconciler) populateEventsURL(log logrus.FieldLogger, agent *aiv1b
 	return nil
 }
 
+func generateControllerLogsDownloadURL(baseURL string, clusterID string, authType auth.AuthType, filterBy ...string) (string, error) {
+	path := fmt.Sprintf("%s%s/v2/clusters/%s/logs",
+		baseURL, restclient.DefaultBasePath, clusterID)
+	u, err := url.Parse(path)
+	if err != nil {
+		return "", err
+	}
+	queryParams := u.Query()
+	for i := 0; i < len(filterBy)-1; i += 2 {
+		queryParams.Set(filterBy[i], filterBy[i+1])
+	}
+	u.RawQuery = queryParams.Encode()
+	downloadURL := u.String()
+	if authType != auth.TypeLocal {
+		return downloadURL, nil
+	}
+
+	downloadURL, err = gencrypto.SignURL(downloadURL, clusterID, gencrypto.ClusterKey)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to sign cluster controller logs URL")
+	}
+
+	return downloadURL, nil
+}
+
+func areLogsCollected(logCollectionTime strfmt.DateTime) bool {
+	return !time.Time(logCollectionTime).Equal(time.Time{}) // timestamp update, meaning logs were collected from a controller
+}
```

Comment:
We may not need this function, what do you think? We are consuming it from one place only and It's only doing this one thing, which we can likely do inline.

Reply:
Agreed! I'll go ahead and move it to the single line